test/test_quantized.py

```diff
@@ -1089,423 +1088,425 @@ def test_qlinear(self, batch_size, input_channels, output_channels,
         self.assertEqual(Y_fp32, Y_fp32_ref,
                          message="torch.ops.quantized.linear_dynamic (fbgemm) results are off")
 
-@unittest.skipUnless('fbgemm' in torch.backends.quantized.supported_engines,
-                     " Quantized operations require FBGEMM. FBGEMM is only optimized for CPUs"
-                     " with instruction set support avx2 or newer.")
-class TestQuantizedLinear(unittest.TestCase):
-    """Tests the correctness of the quantized linear and linear_relu op."""
-    @no_deadline
-    @given(batch_size=st.integers(1, 4),
-           input_channels=st.integers(16, 32),
-           output_channels=st.integers(4, 8),
-           use_bias=st.booleans(),
-           use_relu=st.booleans(),
-           use_multi_dim_input=st.booleans(),
-           use_channelwise=st.booleans())
```

Comment:
Why not just add another hypothesis parameter here qengine and select "fbgemm" and "qnnpack" for that parameter?

Reply:
Either way works. I took this approach as we need to check other conditions for qnnpack like `if not IS_PPC and not TEST_WITH_UBSAN:` and skip calling the test if either of these conditions are set.

Reply:
We can add those check inside the test itself and make it pass. IMO, without the context manager, it would be easier to read.

Reply:
We would still need the context manager to set the correct backend. But I guess it can be moved inside the test.